pkg/controller/instanceset/reconciler_status.go

```diff
@@ -71,17 +73,46 @@ func (r *statusReconciler) Reconcile(tree *kubebuilderx.ObjectTree) (*kubebuilde
 	notReadyNames := sets.New[string]()
 	notAvailableNames := sets.New[string]()
 	currentRevisions := map[string]string{}
+
+	template2TemplatesStatus := map[string]*workloads.InstanceTemplateStatus{}
+	template2TotalReplicas := map[string]int32{}
+	templatesTotalReplicas := int32(0)
+	for _, template := range its.Spec.Instances {
+		templateReplicas := int32(1)
+		if template.Replicas != nil {
+			templateReplicas = *template.Replicas
+		}
+		template2TotalReplicas[template.Name] = templateReplicas
+		templatesTotalReplicas += templateReplicas
+	}
+	if *its.Spec.Replicas > templatesTotalReplicas {
+		template2TotalReplicas[""] = *its.Spec.Replicas - templatesTotalReplicas
```

Comment:
A template with an empty name ('') is confusing. Should we change it to a more readable name like 'default'? We could also add a restriction that no template should be named 'default'. @free6om

Reply:
Yes, instead of leaving it empty, it's better to have a default name.

Reply:
I'd prefer to **NOT** display the `default` Template's status in the `TemplatesStatus` field:
1. `status.templatesStatus` should be a direct observation of the `spec.instances`, and the `default` template is not allowed to specified in the `spec.instances` field, it should not be observed through `status.templatesStatus` either. 
2. one Template is enough for most cases.  this means `spec.instances` is `nil`, then the`status.templatesStatus` should be `nil` too: I don't care `spec.instances`, I don't and shouldn't care `status.templatesStatus` either.

Reply:
> > I'd prefer to **NOT** display the `default` Template's status in the `TemplatesStatus` field:
> > 
> > 1. `status.templatesStatus` should be a direct observation of the `spec.instances`, and the `default` template is not allowed to specified in the `spec.instances` field, it should not be observed through `status.templatesStatus` either.
> > 2. one Template is enough for most cases.  this means `spec.instances` is `nil`, then the`status.templatesStatus` should be `nil` too: I don't care `spec.instances`, I don't and shouldn't care `status.templatesStatus` either.
> 
> @free6om Sounds reasonable, but in the case where the `default` Template and `spec.instances` coexist, do we have a place to intuitively display the details under the `default` Template?

No. the `default` Template is implicit, means the total number of replicas that belong to the `default` Template is calculated by `spec.replicas - sum of(spec.instances[x].replicas)`. The status of the `default` Template can be derived by the same way.

Reply:
> > > I'd prefer to **NOT** display the `default` Template's status in the `TemplatesStatus` field:
> > > 
> > > 1. `status.templatesStatus` should be a direct observation of the `spec.instances`, and the `default` template is not allowed to specified in the `spec.instances` field, it should not be observed through `status.templatesStatus` either.
> > > 2. one Template is enough for most cases.  this means `spec.instances` is `nil`, then the`status.templatesStatus` should be `nil` too: I don't care `spec.instances`, I don't and shouldn't care `status.templatesStatus` either.
> > 
> > 
> > @free6om Sounds reasonable, but in the case where the `default` Template and `spec.instances` coexist, do we have a place to intuitively display the details under the `default` Template?
> 
> No. the `default` Template is implicit, means the total number of replicas that belong to the `default` Template is calculated by `spec.replicas - sum of(spec.instances[x].replicas)`. The status of the `default` Template can be derived by the same way.

I see, the `default` template implicitly sounds reasonable.